website/pages/docs/internals/limits.mdx

```diff
@@ -0,0 +1,271 @@
+| Metadata value size | 512 bytes |
+
+Vault shards the entities across 256 storage entries. This creates a
+hard limit of 128MiB storage space used for entities on Consul, or
+256MiB on integrated storage with its default settings. Entity aliases
+are stored inline in the Entity objects and so consume the same pool
+of storage. Entity definitions are compressed within each storage
+entry, and the pre-compression size varies with the number of entity
+aliases and the amount of metadata.  Minimally-populated entities
+about 200 bytes after compression.
+
+Group definitions are stored separately, in their own pool of 256
+storage entries. The size of each group object depends on the number
+of members and the amount of metadata.  Group aliases and group
+membership information is stored inline in each Group object. A group
+with no metadata, holding 10 entities, will use about 500 bytes per
+group.  A group holding 100 entities would instead consume about 4,000
+bytes.
+
+The following table shows a best-case estimate and a more conservative
+estimate for entities and groups. The number is slightly less than the
+amount that fits in one shard, to reflect the fact that the first
+shard to fill up will start inducing failures.  This maximum will
+decrease if each entity has a large amount of metadata, or if each
+group has a large number of members.
+
+|   | Consul default (512 KiB) | Integrated storage default (1 MiB) |
+|---|--------------------------|------------------------------------|
+| Maximum number of identity entities (best case, 200 bytes per entity) | ~610,000 | ~1,1250,000 |
+| Maximum number of identity entities (conservative case, 500 bytes per entity) | ~250,000 | ~480,000 | 
+| Maximum number of groups (10 entities per group) | ~250,000 | ~480,000 |
+| Maximum number of groups (100 entities per group) | ~22,000 | ~50,000 |
+| Maximum number of members in a group | ~11,500 | ~23,000 |
+ 
+The number of entities can be monitored using Vault's [telemetry](/docs/internals/telemetry#token-identity-and-lease-metrics); see `vault.identity.num_entities` (total) or `vault.identity.entities.count` (by namespace).
+
+The cost of entity and group updates grows as the number of objects in
+each shard increases. This cost can be monitored via the
+`vault.identity.upsert_entity_txn` and
+the `vault.identity.upsert_group_txn` metrics.
+
+Very large internal groups should be avoided (more than 1000 members),
+because the membership list in a group must reside in a single storage entry.
+Instead, consider using [external groups](/docs/secrets/identity#external-vs-internal-groups) or split the group up into multiple sub-groups.
+
+### Token limits
+
+One storage entry is used per token; there is thus no
+upper bound on the number of active tokens.  There are no restrictions on
+the token metadata field, other than the entire token must fit into one
+storage entry:
+
+|                                       |          | 
```

Comment:
Same here... The table may look better with some sort of header.

```suggestion
|            |   General limit   |
```

Reply:
I added just "Limit" as a header.